### ISSUE TITLE
False-positive warning with assigning nil to userdata object

### ISSUE THREAD

**Issue by ligurio** [EmmyLuaLs/emmylua-analyzer-rust#1169]
Imagine the following code with defining a variable with userdata and then assigning `nil` to the variable.
emmylua_check thinks it is not possible to assign nil to userdata, I suppose it is a false positive warning, there is nothing wrong with this assigning.

```
$ cat /tmp/x.lua 
local fin = newproxy(true)
fin = nil
```

```
$ emmylua_check /tmp/x.lua
<snipped>

warning: Cannot assign `nil` to `userdata`.  [assign-type-mismatch]
  -->: :2:1

  1 │ local fin = newproxy(true)
  2 │ fin = nil

<snipped>
```

Version: emmylua_check 0.22.0

**Comment by CppCXY** [comment]
No, this is correct. Keeping variables always non-nullable is an important static check. If you need it to be nullable, you should use ---@type userdata?

**Comment by ligurio** [comment]
Thanks! This works:

```lua
---@type userdata?
local fin = newproxy(true)
fin = nil
```
